platform/xbox/functions/math/pow.c

```diff
@@ -2,15 +2,48 @@
 
 double pow(double x, double y)
 {
-    return exp(y*log(x));
+    if (x < 0.0 && y != trunc(y)) {
+        // Negative base and non-integer exponent
+        return -NAN;
+    }
+
+    float r = exp(y * log(fabs(x)));
+
+    if (x < 0.0 && remainder(y, 2.0) != 0) {
+        r = copysign(r, x);
+    }
+
+    return r;
 }
 
 float powf(float x, float y)
 {
-    return expf(y*logf(x));
+    if (x < 0.0f && y != truncf(y)) {
+        // Negative base and non-integer exponent
+        return -NAN;
+    }
+
+    float r = expf(y * logf(fabsf(x)));
+
+    if (x < 0.0f && remainderf(y, 2.0f) != 0) {
+        r = copysignf(r, x);
```

Comment:
`copysign` shouldn't be necessary, because we already made sure that `x` is negative.
So `r = -r` should always be the same, but potentially faster.

Or was this done to handle NaN or Inf or something in a special way?

Reply:
Ah, this should indeed be simplified. Done.